napari/benchmarks/benchmark_shapes_layer.py

```diff
@@ -27,48 +27,140 @@
     from napari.benchmarks.utils import Skip
 
 
-class Shapes2DSuite:
+class BackendType(Enum):
+    bermuda = (
+        auto()
+    )  # data processing using PartSegCore, ready to switch to bermuda
+    triangle = auto()  # data preprocessing using numba, edge triangulation using numba, triangulation using triangle
+    numba = auto()  # data preprocessing and edge triangulation using numba, triangulation using vispy
+    pure_python = auto()  # data preprocessing, edge triangulation, and triangulation using python and vispy
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        return self.name
+
+
+backend_list = list(BackendType)
+
+
+class _BackendSelection:
+    triangulate: Callable | None
+    prev_settings: bool
+    prev_numba: dict[str, Callable]
```

Comment:
maybe a comment on each of these class attributes?

Reply:
Added description. Could you check?